server/src/main/java/org/tctalent/server/service/db/impl/CandidateServiceImpl.java

```diff
@@ -2624,10 +2624,7 @@ public Candidate getCandidateFromRequest(@Nullable Long requestCandidateId) {
                     .orElseThrow(() -> new NoSuchObjectException(Candidate.class, requestCandidateId));
         } else {
             // Coming from Candidate Portal
-            candidate = authService.getLoggedInCandidate();
-            if (candidate == null) {
-                throw new InvalidSessionException("Not logged in");
-            }
+           candidate = getLoggedInCandidate().orElseThrow(() -> new InvalidSessionException("Not logged in"));
```

Comment:
This getLoggedInCanddiate() method was already in the code and basically does what we want, in that it uses the loggedIn object to get an ID which we can then fetch the updated object with. So this is used to replace where we used to have authService.getLoggedInCandidate() which just used the stored candidate object that belonged to the loggedInUser object.